pike/model.py

```diff
@@ -1870,21 +1870,25 @@ def get_symlink(self, file):
         return self.connection.transceive(
                 self.get_symlink_request(file).parent.parent)[0]
 
-    def enumerate_snapshots_request(self, fh, max_output_response=16384):
+    def enumerate_snapshots_request(self, fh,
+                                    snap_request=smb2.EnumerateSnapshotsRequest,
```

Comment:
Is this an attempt to overload the `enumerate_snapshots_request` to pass the SetSparse/ZeroData requests?

I think it would be better to create an `ioctl_reqest` helper that sets up the `IoctlRequest` with a buffer size, etc, and then have separate helper methods for `set_sparse_request` and `zero_data_request`.

Reply:
Done